### PR TITLE
Architecture overview

### PR DESCRIPTION
**Pull request by neeraj-laad**
Adds text describing the high-level architecture components of Appsody and how they help developers create applications.

## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [ ] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Related Issues
Relates to https://github.com/appsody/website/pull/431

### CODE REVIEW

File: content/docs/overview.md [hunk 1]
```diff
@@ -25,9 +25,17 @@ No project is complete without a nice new CLI to play with. The [Appsody CLI](/c
...
+Appsody simplifies the creation of cloud-native applications and helps you throughout the development lifecycle.
 
-When developers run, debug, or test their application using the Appsody CLI, it starts a container with the stack image, makes the development workspace available to the running container, and starts the Appsody controller. The controller is configurable through environment variables in the stack and manages the application within the running container. For example, the controller can watch for changes in the application `/src` directory. When these changes are saved, the controller restarts the application running in a docker container.
+Appsody provides pre-configured application stacks, which specify which language runtimes, frameworks, libraries, and tools that you want to use. Appsody stacks abstract various degrees of infrastructure and application concerns and provide consistency and maintainability across your deployments. Project templates build upon these stacks, providing developers with a starter application to bring into their IDE of choice and begin development.
```
Comment (helenmasters) [inline content/docs/overview.md]: I've done some rewording.... you may disagree with some but I didn't like the "abstract various degrees of...."

I still don't like the "pull down" reference as it sounds like a pull down menu - but that's not what it means, can you think of another word to use instead - maybe "create the stack within your development environment"?

-----

"Appsody simplifies the creation of cloud-native applications and helps you throughout the development lifecycle.

Appsody provides preconfigured application stacks, which specify the language runtimes, frameworks, libraries, and tools you want to use. Appsody stacks abstract infrastructure and application concerns and provide consistency and maintainability across your deployments. Project templates build upon these stacks, providing developers with a starter application to bring into their IDE of choice and begin development.

You can use the Appsody CLI to discover the preconfigured technology stacks, which provide everything you need to get started building apps. Stacks can be customized and extended to suit you or your teams use cases and standards. Using stacks can be a great way to enable new developers in your team to become productive faster.

After you choose the stack that you want to work with, you can get started by using the `init` command to pull down the stack to your development environment.

Appsody provides an inner loop development experience. When developers `run`, `debug`, or `test` their application by using the Appsody CLI it starts a container with the stack image, makes the development workspace available to the running container, and starts the application. The Appsody controller watches for file changes and manages live updates to your application within the running container. 

When your application is ready, you can `build` it into a production-ready image and push the image to a registry. You can also `deploy` the application to a local, or remote Kubernetes cluster."

Comment (neeraj-laad) [reply content/docs/overview.md]: After you choose the stack that you want to work with, you can initialize your development environment with the stack by using the `init` command.

or 

After you choose the stack that you want to work with, you can get started by using the `init` command to initialize your development environment with the stack.

Comment (SueChaplain) [reply content/docs/overview.md]: How about "pull a container that contains the stack to your development environment."? Most users would, I think, be able to associate the pull term with containers and avoid confusion.

Comment (neeraj-laad) [reply content/docs/overview.md]: rest of your changes look ok to me.

Comment (helenmasters) [reply content/docs/overview.md]: Neeraj and Sue, any of your suggested wording changes sound better to me!

Comment (neeraj-laad) [reply content/docs/overview.md]: I'm ok with any of them as well..  We should say `pull image` or `pull container image` if we go with that option.

**Pull request merged**